### PR TITLE
[ch27976] Fix bug on update one for spot check

### PR DESCRIPTION
**Pull request by emaciupe**

### CODE REVIEW

File: src/etools/applications/partners/models.py [hunk 1]
```diff
@@ -797,41 +797,52 @@ def update_programmatic_visits(self, event_date=None, update_one=False):
...
-        return SpotCheck.objects.filter(
+        audit_sc = SpotCheck.objects.filter(
             partner=self, date_of_draft_report_to_ip__year=datetime.datetime.now().year
         ).exclude(status=Engagement.CANCELLED)
 
+        t2f_sc = TravelActivity.objects.filter(
+            travel_type=TravelType.SPOT_CHECK,
+            travels__status=Travel.COMPLETED,
+            date__year=datetime.datetime.now().year,
+            partner=self
```
Comment (domdinicola) [inline src/etools/applications/partners/models.py:808]: I think spot checks from T2F are archived (we don't count anymore)

File: src/etools/applications/partners/models.py [hunk 2]
```diff
@@ -797,41 +797,52 @@ def update_programmatic_visits(self, event_date=None, update_one=False):
...
-        self.save()
+            audit_sc = self.spot_checks['audit_sc']
+            t2f_sc = self.spot_checks['t2f_sc']
+
+            a_sc1 = audit_sc.filter(date_of_draft_report_to_ip__quarter=1).count()
+            a_sc2 = audit_sc.filter(date_of_draft_report_to_ip__quarter=2).count()
+            a_sc3 = audit_sc.filter(date_of_draft_report_to_ip__quarter=3).count()
+            a_sc4 = audit_sc.filter(date_of_draft_report_to_ip__quarter=4).count()
+
+            t2f_sc1 = t2f_sc.filter(date__quarter=1).count()
```
Comment (domdinicola) [inline src/etools/applications/partners/models.py:835]: pls remove

**Review by domdinicola — CHANGES_REQUESTED** [review]

File: src/etools/applications/partners/models.py (hunk 1 above)
Comment (emaciupe) [reply src/etools/applications/partners/models.py:808]: @domdinicola From what I see in t2f.models line 280, spot checks are being counted.

Comment (emaciupe) [reply src/etools/applications/partners/models.py]: resolved

File: src/etools/applications/partners/models.py (hunk 2 above)
Comment (emaciupe) [reply src/etools/applications/partners/models.py]: resolved

File: src/etools/applications/audit/models.py [hunk 3]
```diff
@@ -456,7 +456,7 @@ def submit(self, *args, **kwargs):
     @transition('status', source=Engagement.STATUSES.report_submitted, target=Engagement.STATUSES.final,
                 permission=has_action_permission(action='finalize'))
     def finalize(self, *args, **kwargs):
-        self.partner.update_spot_checks(update_one=True, event_date=self.date_of_draft_report_to_ip)
+        self.partner.update_spot_checks(event_date=self.date_of_draft_report_to_ip)
```
[the remark this answers is not in the document]
Comment (domdinicola) [reply src/etools/applications/audit/models.py:459]: @robertavram I guess we can remove this line (no need to recalculate either)

[the remark this answers is not in the document]
Comment (domdinicola) [reply src/etools/applications/audit/models.py:459]: remove this line

[the remark this answers is not in the document]
Comment (emaciupe) [reply src/etools/applications/audit/models.py]: solved

**Review by domdinicola — APPROVED** [review]

File: src/etools/applications/partners/models.py [hunk 4]
```diff
@@ -828,10 +825,8 @@ def update_spot_checks(self, event_date=None, update_one=False):
             self.hact_values['spot_checks']['completed']['q3'] = asc3
             self.hact_values['spot_checks']['completed']['q4'] = asc4
 
-            sc = audit_spot_check.count()  # TODO 1.1.9c add spot checks from field monitoring
-
-        self.hact_values['spot_checks']['completed']['total'] = sc
-        self.save()
+            self.hact_values['spot_checks']['completed']['total'] = audit_spot_check.count()  # TODO 1.1.9c add spot checks from field monitoring
+        self.save(update_fields=['hact_values'])
```
Comment (robertavram) [inline src/etools/applications/partners/models.py:829]: nice one!

**Pull request merged**